code/datums/components/laser_components.dm

```diff
@@ -0,0 +1,27 @@
+/datum/component/extralasers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/obj/item/external_lens/lens
+	var/obj/item/ammo_casing/energy/laser/ammo
+
+/datum/component/extralasers/Initialize(ammo, _lens)
+	if(!istype(parent, /obj/item/gun/energy/laser))
+		return COMPONENT_INCOMPATIBLE
+	var/obj/item/gun/energy/laser/L = parent
+	L.ammo_type  += new ammo (src)
+	lens = _lens
+	qdel(_lens)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/detach)
+
+/datum/component/extralasers/proc/detach(datum/source, obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_CROWBAR)
+		var/obj/item/gun/energy/laser/L = parent
+		L.chambered = null
+		var/obj/item/ammo_casing/energy/R = LAZYACCESS(L.ammo_type, L.ammo_type.len)
+		LAZYREMOVE(L.ammo_type, R)
+		L.select_fire(user)
+		L.recharge_newshot()
+		L.update_icon(TRUE)
+		RemoveComponent()
+		var/turf/T = user.loc
+		new ammo(T) //doest work
```

Comment:
@ninjanomnom for some reason it gives the error "cant create object of obj/item/external_lens/type"

Reply:
Doesn't look like you ever assign to the ammo var

Reply:
uhh i think the problem was lens was a reference to a deleted object and new wants a path to the object
this new thing gives the runtime that it fired COMPONENT INCOMPATIBLE on  the mob and idk why

Reply:
Because you're trying to apply the component to the wrong type